migrations/V300__transactions_count.sql

```diff
@@ -0,0 +1,11 @@
+ALTER TABLE payload
+    ADD COLUMN num_transactions INTEGER;
+
+-- Initialize the `num_transactions` column by counting transactions for each
+-- existing payload.
+UPDATE payload AS p
```

Comment:
This seems like an expensive query, and we have over 1 million transactions on the mainnet. I'm not sure how long this will take.
We can also do a temporary table to store transaction counts and then update?

Reply:
I think it will be a bit slow but we've run slow migrations before (ie dropping the leaf payloads). It just means the node takes a while to start up initially, but since migrations run before anything else, it doesn't interfere with the performance of the node after it's started.

For reference I tried the following read-only equivalent on Decaf (800k blocks, 900k transactions)
```sql
explain analyze select p.height, (select count(*) from transaction as t where t.block_height = p.height) as num_transactions from payload as p;
                                                                      QUERY PLAN                                                                      
------------------------------------------------------------------------------------------------------------------------------------------------------
 Index Only Scan using payload_pkey on payload p  (cost=0.42..7055578.39 rows=804740 width=16) (actual time=1.453..2877.085 rows=816574 loops=1)
   Heap Fetches: 99568
   SubPlan 1
     ->  Aggregate  (cost=8.73..8.74 rows=1 width=8) (actual time=0.003..0.003 rows=1 loops=816574)
           ->  Index Only Scan using transaction_pk on transaction t  (cost=0.55..8.71 rows=9 width=0) (actual time=0.002..0.002 rows=1 loops=816574)
                 Index Cond: (block_height = p.height)
                 Heap Fetches: 7913
 Planning Time: 3.260 ms
 Execution Time: 2921.088 ms
(9 rows)

```

Expensive for sure but not unreasonable. Notice it only looks at indexes. The update will of course be much slower since writing is slower than reading, but even if it's 100 times slower, the migration would complete in about 5 minutes.

Can you say more about why a temporary table would speed things up?

Reply:
Lol I just ran the actual migration on decaf query-db0 (wrapped in begin/rollback). It took 11 seconds. So I think we are good

Reply:
Okay, perfect! I just wasn’t sure how long it would take. 
I was thinking of creating a new temp table with a GROUP BY on transactions.height, then creating an index on height in that temp table, and finally running the migration with a JOIN on height.

Reply:
11s wow!

Reply:
Basically, I thought it would run a count query for n blocks, which I expected to be slower than  a GROUP BY and  join on update, but now I’m not so sure, haha.